lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,344 @@
+
+logger = logging.getLogger('TwoWaySMSv2')
+logger.setLevel(logging.getLevelName(LOG_LEVEL))
+
+################################################################################################
+# Get the database URI from SSM Parameter Store.
+################################################################################################
+
+if os.getenv("NOTIFY_ENVIRONMENT") == "test":
+    sqlalchemy_database_uri = os.getenv("SQLALCHEMY_DATABASE_URI")
+else:
+    database_uri_path = os.getenv("DATABASE_URI_PATH")
+    if database_uri_path is None:
+        # Without this value, this code cannot know the path to the required
+        # SSM Parameter Store resource.
+        sys.exit("DATABASE_URI_PATH is not set.  Check the Lambda console.")
+
+    logger.debug("Getting the database URI from SSM Parameter Store . . .")
+    ssm_client = boto3.client("ssm", region_name=AWS_REGION)
+    ssm_response: dict = ssm_client.get_parameter(
+        Name=database_uri_path,
+        WithDecryption=True
+    )
+    logger.debug(". . . Retrieved the database URI from SSM Parameter Store.")
+    sqlalchemy_database_uri = ssm_response.get("Parameter", {}).get("Value")
+
+    if sqlalchemy_database_uri is None:
+        sys.exit("Can't get the database URI from SSM Parameter Store.")
+
+################################################################################################
+# Use the database URI to get a 10DLC-to-URL mapping from the database.
+# The format should be:
+#    {
+#        'number': {
+#            'service_id': <value>,
+#            'url_endpoint': <value>,
+#            'self_managed': <value>,
+#        }
+#    }
+################################################################################################
+
+if os.getenv("NOTIFY_ENVIRONMENT") == "test":
+    two_way_sms_table_dict = {}
+else:
+    try:
+        logger.info("Retrieving the 10DLC-to-URL mapping from the database . . .")
+        db_connection = psycopg2.connect(sqlalchemy_database_uri)
+        if db_connection is None:
+            sys.exit("No database connection.")
+
+        with db_connection.cursor() as c:
+            # https://www.psycopg.org/docs/cursor.html#cursor.execute
+            c.execute(INBOUND_NUMBERS_QUERY)
+            mapping_data = c.fetchall()
+            logger.debug("Data returned from query: %s", mapping_data)
+
+        db_connection.close()
+        logger.info(". . . Retrieved the 10DLC-to-URL mapping from the database.")
+    except psycopg2.Warning as e:
+        logger.warning(e)
+    except psycopg2.OperationalError as e:        
+        logger.exception(e)
+        sys.exit("Unable to retrieve the 10DLC-to-URL mapping from the database.")
+
+    # Create the mapping table with a generator expression.
+    two_way_sms_table_dict = {
+        n: {
+            'service_id': s,
+            'url_endpoint': u,
+            'self_managed': False if sm == None else sm,
```

Comment:
I don't understand how that could be `None`? The field was set to [not nullable](https://github.com/department-of-veterans-affairs/notification-api/blob/master/app/models.py#L514), with a default of `False`. Not tracking where `None` can work it's way in there.

Reply:
I removed the None check.